### PR TITLE
created the notification component.

### LINKED ISSUE #695: Feedback Notification component
refer https://github.com/starknet-id/starknet.quest/issues/693

### PR DESCRIPTION
**Pull request by Dprof-in-tech**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature: created the notification component with the needed message props.

Resolves: #695 

## Other information

below is an image of the implementation using a button
image

## Summary by CodeRabbit

- **New Features**
  - Introduced a notification system, including a `NotificationProvider` component and `useNotification` hook, for managing and displaying notifications (success, error, info, warning) with customizable messages and auto-hide durations.

- **Improvements**
  - Enhanced the `Providers` component by integrating the new `NotificationProvider` for better notification management.
  - Simplified data structures passed to the `Stats` component for improved performance and readability.

### CODE REVIEW

**Comment by ayushtom** [comment]
hey
i would require you to make a context provider where i can call a hook called showNotification("the message", type="success")
and the toast just shows up on the screen where this function is called.And in this you can optionally add the onOpen and onClose params in this function.

I think importing the component is a bit repetitive.

**Review by ayushtom — CHANGES_REQUESTED** [review]
Mentioned above

**Comment by Dprof-in-tech** [comment]
Screenshot from 2024-05-29 06-36-10
Screenshot from 2024-05-29 06-36-44
Screenshot from 2024-05-29 06-36-55


These are sample images of what the various feedback look like right now
corrections are welcome
Thank you ser.

File: app/page.tsx [hunk 1]
```diff
@@ -30,49 +35,28 @@ export default function Page() {
...
+            { name: "Quests NFT minted", value: "+1M" },
+            { name: "Unique addresses", value: "398K" },
+            { name: "Unique visitors", value: "+200K" },
           ]}
         />
         <div className={styles.blur2}>
           <Blur green />
         </div>
         <HowToParticipate />
+        {/* <button onClick={handleClick}>Show Success Notification</button> */}
```
Comment (ayushtom) [inline app/page.tsx]: can remove this

File: package-lock.json
Comment (ayushtom) [inline package-lock.json:1]: do we install any new packages? 
i dont think we do so we can remove this change

**Review by ayushtom — COMMENTED** [review]
The logo color can be changed for info and error
the notification bar looks good in black

Maybe use a yellow info icon for info 
and red cross icon for error

File: package-lock.json
Comment (ayushtom) [inline package-lock.json:1]: please remove this file since nothing in changes in packages

Comment (Dprof-in-tech) [reply package-lock.json:1]: I already deleted those sections or do I still delete the file cos it would be rebuilt while being deployed

**Review by ayushtom — APPROVED** [review]

**Pull request merged**